contrib_bots/lib/poll.py

```diff
@@ -0,0 +1,105 @@
+
+
+class PollHandler(object):
+    '''
+    This plugin facilitates polling.
+    '''
+
+    def __init__(self):
+        # polls is a dict of dict. First dict corresponds to poll title,
+        # second refers to answers in poll
+        self.polls = dict()
+
+        # streams is a convenience object that keeps track of the latest
+        # question for each stream.
+        self.streams = dict()
+
+    def usage(self):
+        return '''
+        This plugin facilitates polling.
+
+        Commands:
+        @question poll title: option 1, option 2, option 3
+        A valid poll must use a colon (:) to devide question from answers.
+        Answers must be seperate by coma (,).
+
+        @answer option 1
+        Question will be infered by scanning channel for last poll with
+        valid answer.
+
+        @result poll title
+        Returns a formatted version of the poll.
+
+        >> @question what's the best ice cream flavour: vanilla, chocolate
+        << Question successfully created! Answer away
```

Comment:
I think repeating the poll on the channel would be useful here - the `@question` message might be not next to the response `Question successfully created! ...` and the users could get confused as to what was the question and options.

Could you make this response a bit clearer? Such as:
```
@<author> created a new poll!

Question: <question> # currently <poll_title>
option 1: <option_1>
option 2: <option_2>
etc.

You can answer this poll by writing `@answer <option>`.
```